libkrun/libkruntime.c

```diff
@@ -342,126 +419,213 @@ libkruntime_done(void)
 }
 
 uint64_t
-read_core_cycles(void)
+krun_read_core_cycles(int core)
 {
 #if defined(__linux__) && !defined(TRAVIS)
-    return sum_msr_all_cores(MSR_IA32_PERF_FIXED_CTR1, pctr_val_mask);
+    return krun_read_msr(core, MSR_IA32_PERF_FIXED_CTR1) & krun_pctr_val_mask;
 #elif defined(__linux__) && defined(TRAVIS)
     return 0;
 #else
 #error "Unsupported platform"
 #endif
 }
 
-uint64_t
-read_aperf(void)
+static uint64_t
+krun_read_aperf(int core)
 {
 #if defined(__linux__) && !defined(TRAVIS)
-    return sum_msr_all_cores(IA32_APERF, IA32_APERF_MASK);
+    return krun_read_msr(core, IA32_APERF) & IA32_APERF_MASK;
 #elif defined(__linux__) && defined(TRAVIS)
     return 0;
 #else
 #error "Unsupported platform"
 #endif
 }
 
-uint64_t
-read_mperf(void)
+static uint64_t
+krun_read_mperf(int core)
 {
 #if defined(__linux__) && !defined(TRAVIS)
-    return sum_msr_all_cores(IA32_MPERF, IA32_MPERF_MASK);
+    return krun_read_msr(core, IA32_MPERF) & IA32_MPERF_MASK;
 #elif defined(__linux__) && defined(TRAVIS)
     return 0;
 #else
 #error "Unsupported platform"
 #endif
 }
 
-#if defined(__linux__) && !defined(TRAVIS)
 /*
- * Sum the values of an MSR across all cores.
+ * Check for double precision loss.
```

Comment:
This comment could do with tightening and shortening: it's a bit repetitive at the moment. I'd like to keep the technical info in, though, which really helps readers.